app/components/network_indicator/network_indicator.js

```diff
@@ -0,0 +1,390 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import {intlShape} from 'react-intl';
+import {
+    ActivityIndicator,
+    Alert,
+    Animated,
+    AppState,
+    Platform,
+    StyleSheet,
+    TouchableOpacity,
+    View,
+} from 'react-native';
+import DeviceInfo from 'react-native-device-info';
+import IonIcon from 'react-native-vector-icons/Ionicons';
+
+import FormattedText from 'app/components/formatted_text';
+import {ViewTypes} from 'app/constants';
+import mattermostBucket from 'app/mattermost_bucket';
+import networkConnectionListener, {checkConnection} from 'app/utils/network';
+import {t} from 'app/utils/i18n';
+import LocalConfig from 'assets/config';
+
+import {RequestStatus} from 'mattermost-redux/constants';
+
+const HEIGHT = 38;
+const OFFLINE = 'offline';
+const CONNECTING = 'connecting';
+const CONNECTED = 'connected';
+const {
+    ANDROID_TOP_LANDSCAPE,
+    ANDROID_TOP_PORTRAIT,
+    IOS_TOP_LANDSCAPE,
+    IOS_TOP_PORTRAIT,
+    IOSX_TOP_PORTRAIT,
+    STATUS_BAR_HEIGHT,
+} = ViewTypes;
+
+export default class NetworkIndicator extends PureComponent {
+    static propTypes = {
+        actions: PropTypes.shape({
+            closeWebSocket: PropTypes.func.isRequired,
+            connection: PropTypes.func.isRequired,
+            initWebSocket: PropTypes.func.isRequired,
+            logout: PropTypes.func.isRequired,
+            startPeriodicStatusUpdates: PropTypes.func.isRequired,
+            stopPeriodicStatusUpdates: PropTypes.func.isRequired,
+        }).isRequired,
+        isConnecting: PropTypes.bool,
+        isLandscape: PropTypes.bool,
+        isOnline: PropTypes.bool,
+        webSocketStatus: PropTypes.string,
+    };
+
+    static defaultProps = {
+        isOnline: true,
+    };
+
+    static contextTypes = {
+        intl: intlShape.isRequired,
+    };
+
+    static getDerivedStateFromProps(nextProps, prevState) {
+        const {isLandscape, networkStatus} = prevState;
+
+        if (nextProps.isLandscape !== isLandscape && networkStatus) {
+            const navBar = this.getNavBarHeight(nextProps.isLandscape);
+            const top = new Animated.Value(navBar - HEIGHT);
+
+            return {
+                navBar,
+                top,
+            };
+        }
+
+        return null;
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.isX = DeviceInfo.getModel().includes('iPhone X');
```

Comment:
this is a constant now for this `IS_IPHONE_X` in `constants/device`

Reply:
Done